### PR TITLE
fix: rescheduling dynamic booking

### LINKED ISSUE #10554: Dynamic Event Rescheduling has Disabled && Mandatory fields
Found a bug? Please fill out the sections below. 👍

### Issue Summary

Dynamic Event Rescheduling has Disabled && Mandatory fields

### Steps to Reproduce
1. Setup dynamic event with +
2. Try to reschedule 

### Actual Results

- Couldnt reschedule bc disabled fields are mandatory

### Expected Results

- Let me reschedule

### Evidence

grafik

https://app.cal.com/kristian-mikhel-o0bkzq+johannes/dynamic?rescheduleUid=5zdpZAyNAPALjRtvCdo3a8&duration=60&date=2023-08-10&month=2023-08&slot=2023-08-10T12%3A00%3A00.000Z

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

When rescheduling a dynamic booking, we were relying on `eventTypeId` to get the booking data, which for dynamic bookings it doesn't exist. Not we rely on the event type entirely, fixing the issue.

Fixes #10554 

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

Reschedule a dynamic booking and see you are correctly placed in the available slots page to be able to reschedule.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/features/bookings/lib/get-booking.ts [hunk 1]
```diff
@@ -109,24 +109,20 @@ export const getBookingWithResponses = <
...
-    (
-      await prisma.booking.findFirst({
-        where: {
-          uid,
-        },
-        select: {
-          eventTypeId: true,
-        },
-      })
-    )?.eventTypeId || null;
```
Comment (leog) [inline packages/features/bookings/lib/get-booking.ts:124]: `eventTypeId` is set to null for dynamic bookings, as they don't respond to any event type, which made the booking data to be null, triggering an odd behavior in the booker state

File: packages/features/bookings/lib/get-booking.ts [hunk 2]
```diff
@@ -109,24 +109,20 @@ export const getBookingWithResponses = <
...
+    select: {
+      id: true,
+    },
+  });
 
   // If no booking is found via the uid, it's probably a booking seat,
   // which we query next.
   let attendeeEmail: string | null = null;
-  if (!eventTypeId) {
+  if (!eventType) {
```
Comment (leog) [inline packages/features/bookings/lib/get-booking.ts:125]: We now rely on the existence of the event type entirely.

File: packages/features/bookings/lib/get-booking.ts [hunk 3]
```diff
@@ -109,24 +109,20 @@ export const getBookingWithResponses = <
...
-      await prisma.booking.findFirst({
-        where: {
-          uid,
-        },
-        select: {
-          eventTypeId: true,
-        },
-      })
-    )?.eventTypeId || null;
+  const eventType = await prisma.booking.findFirst({
```
Comment (zomars) [inline packages/features/bookings/lib/get-booking.ts:113]: Shouldn't this be booking then?

```suggestion
  const booking = await prisma.booking.findFirst({
```

Comment (leog) [reply packages/features/bookings/lib/get-booking.ts]: yeahp, thanks!

File: packages/features/bookings/lib/get-booking.ts [hunk 4]
```diff
@@ -109,24 +109,20 @@ export const getBookingWithResponses = <
...
-      await prisma.booking.findFirst({
-        where: {
-          uid,
-        },
-        select: {
-          eventTypeId: true,
-        },
-      })
-    )?.eventTypeId || null;
+  const theBooking = await prisma.booking.findFirst({
```
Comment (alannnc) [inline packages/features/bookings/lib/get-booking.ts:113]:
```suggestion
  const booking = await prisma.booking.findFirst({
```

**Review by zomars — APPROVED** [review]
Tested and working. nice job @leog 🙏🏽

**Pull request merged**